### DISCUSSION TITLE
Multi tenants, multi apps, multi env

### DISCUSSION THREAD

**Discussion by redaqafssaoui** [fluxcd/flux2#2485] [General] [answered]
Hello,
I'm studying how to implement Gitops using fluxv2 in my context, I have multiple projects, each project has multiple apps, and those apps are deployed in multiple namespaces ( dev-namespace, staging-namespace, and prod-namespace) in the same cluster k8s ( shared ). In my case I should isolate tenants. Do you have any proposals for this use case?

**Comment by stefanprodan** [comment, marked as answer]
Please see https://github.com/fluxcd/flux2-multi-tenancy This repository serves as a starting point for managing multi-tenant clusters with Git and Flux v2.
